bin/main.ml

```diff
@@ -1,5 +1,6 @@
 open Tezos_context_hash
 open Encoding
+module Node = Node.Make (Hash) (Path) (Metadata)
```

Comment:
Why is this needed?

Reply:
Because in the following line we have:

```ocaml
module Inter = Irmin_pack.Private.Inode.Make_internal (Conf) (Hash) (Node)
```

And this functor has the following type:

```ocaml
  module Make_internal
      (Conf : Conf.S)
      (H : Irmin.Hash.S)
      (Node : Irmin.Private.Node.S with type hash = H.t) :
    Internal
      with type hash = H.t
       and type Val.metadata = Node.metadata
       and type Val.step = Node.step
```

Did I miss something?

Reply:
Ha I see - this is working with irmin master but not 2.5.3. 

Not sure what should be the policy here -- we probably want a branch for 2.5 as we do on the irmin repo. But I think the master branch should continue to compile with irmin's master.